plugins/info/info_impl.cpp

```diff
@@ -116,9 +117,28 @@ void InfoImpl::process_autopilot_version(const mavlink_message_t &message)
     const char *product_name = product_id_str(autopilot_version.product_id);
     STRNCPY(_product.product_name, product_name, sizeof(_product.product_name) - 1);
 
+    if (is_array_zero(autopilot_version.uid2, sizeof(autopilot_version.uid2))) {
+        std::memcpy(_identification.hardware_uuid,
+                    reinterpret_cast<uint8_t *>(&autopilot_version.uid),
+                    sizeof(autopilot_version.uid));
+    } else {
+        std::memcpy(
+            _identification.hardware_uuid, autopilot_version.uid2, sizeof(autopilot_version.uid2));
```

Comment:
I'd make sure the rest of `hardware_uuid` is set to zero in case `process_autopilot_version` is called several times (mostly a precaution).

Reply:
Good catch.